optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -516,6 +516,58 @@ public boolean hasAnyChainedGenuineVariables() {
         return false;
     }
 
+    public GenuineVariableDescriptor deduceVariableDescriptor(String variableName) {
+        GenuineVariableDescriptor<Solution_> variableDescriptor;
+        if (variableName != null) {
+            variableDescriptor = getGenuineVariableDescriptor(variableName);
+            if (variableDescriptor == null) {
+                throw new IllegalArgumentException("The config (" + this
+                        + ") has a variableName (" + variableName
+                        + ") which is not a valid planning variable on entityClass ("
+                        + getEntityClass() + ").\n"
+                        + buildInvalidVariableNameExceptionMessage(variableName));
+            }
+        } else {
+            Collection<GenuineVariableDescriptor<Solution_>> variableDescriptors = getGenuineVariableDescriptors();
+            if (variableDescriptors.size() != 1) {
+                throw new IllegalArgumentException("The config (" + this
+                        + ") has no configured variableName (" + variableName
+                        + ") for entityClass (" + getEntityClass()
+                        + ") and because there are multiple variableNames ("
+                        + getGenuineVariableNameSet()
+                        + "), it cannot be deduced automatically.");
+            }
+            variableDescriptor = variableDescriptors.iterator().next();
+        }
+        return variableDescriptor;
+    }
+
+    public List<GenuineVariableDescriptor<Solution_>> deduceVariableDescriptorList(List<String> variableNameIncludeList) {
```

Comment:
I'd prefer a no-arg overload to just putting `null` here.

Reply:
will be addressed in the next PR, see the comment below.